test/gtest/parser_unroll.cpp

```diff
@@ -178,12 +178,12 @@ class MidEnd : public PassManager {
 #endif
 
 const IR::P4Parser *getParser(const IR::P4Program *program) {
```

Comment:
Ideally, this should return an error/nullpointer/optional if there is no parser in the program but that is not your responsibility to fix in this PR.

Reply:
I tried to keep the original semantics of the code. Certainly there are lots of BUG_CHECK's here instead of real diagnostics. I think we'd probably need to use something like Expected idiom so the errors could be detected more easily and propagated.